web/src/store/index.ts

```diff
@@ -448,9 +457,11 @@ const actions = {
 
 };
 
-export default new Vuex.Store({
+const store: StoreOptions<RootState> = {
```

Comment:
This is what handles providing type checking to interaction with `store` from within mutations, getters, etc right?

Do you know what happens when you set stuff with `Vue.set(store, 'key', 'val')` ?  I would imagine type validations are just not caught.

I imagine the main purpose of types in vuex is for validating external interactions, such as accessing the store's proerties from `mapState` -- is that right?

Reply:
> This is what handles providing type checking to interaction with `store` from within mutations, getters, etc right?
> 
> Do you know what happens when you set stuff with `Vue.set(store, 'key', 'val')` ? I would imagine type validations are just not caught.
> 

I admit, I don't understand some of the deeper magic in Vue's specific type bindings (there's some magic, for instance, that allows TS to recognize props and computed properties for the value returned by `Vue.extend()`).

An experiment with a poorly typed `Vue.set()` invocation seems to be in order.

> I imagine the main purpose of types in vuex is for validating external interactions, such as accessing the store's proerties from `mapState` -- is that right?

Yes this sounds right (though it is possible that the "internal" interactions might also be checked).